packages/pluggableWidgets/tree-view-web/src/components/hooks/useIncrementalId.tsx

```diff
@@ -0,0 +1,17 @@
+import { useEffect, useState } from "react";
+
+let incrementalId = 0;
+
+/**
+ * A custom hook to create a "unique" id based on an incremental counter that is updated
+ * every time the component is mounted.
+ */
+export const useIncrementalId = (): { id: number } => {
```

Comment:
I am not sure if this will work if you have 2 different pages with same widget ids.. So it will not be unique. Maybe for this incremental ID we should create a global function that gives the next unique id.

Reply:
See [the comment above where it's used](https://github.com/mendix/widgets-resources/pull/838/files/7bf2dd49566f510860a2551168e6f99b74a108c1#diff-7152f69d2d461c0c2fac8144cc93b341a308911fb742168eca5bd00a13025ba0R120-R121), it's only for testing purposes because of the [this line in the useTreeViewFocusChangeHandler](https://github.com/mendix/widgets-resources/pull/838/files/7bf2dd49566f510860a2551168e6f99b74a108c1#diff-51e5617b71e7a57199016e1ab79aec0a2c25c25c02c7636e48f2295c1cf0dfc5R23). In production this is just an empty and unused class, but in testing this is necessary so that the `:scope` part works properly.